### PR TITLE
Add `onchainSigningStrategy` job spec field support

### PR DESCRIPTION
**Pull request by vyzaldysanchez**

### CODE REVIEW

File: .changeset/olive-knives-happen.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"chainlink": patch
+---
+
+#internal Generic Pluging `onchainSigningStrategy` support
```
Comment (cedric-cordenier) [inline .changeset/olive-knives-happen.md:5]: "Generic Plugin"

File: core/services/ocr2/validate/validate.go [hunk 2]
```diff
@@ -173,7 +174,42 @@ func (o *OCR2GenericPluginConfig) UnmarshalJSON(data []byte) error {
...
+	}
+	name, ok := pk.(string)
+	if !ok {
+		return "", fmt.Errorf("expected string publicKey value, but got: %T", pk)
+	}
+	return name, nil
+}
+
 func validateGenericPluginSpec(ctx context.Context, spec *job.OCR2OracleSpec, rc plugins.RegistrarConfig) error {
+	fmt.Println("running validateGenericPluginSpec")
```
Comment (cedric-cordenier) [inline core/services/ocr2/validate/validate.go:212]: Can you remove this logging line?

File: core/services/ocr2/validate/validate.go [hunk 3]
```diff
@@ -188,6 +224,20 @@ func validateGenericPluginSpec(ctx context.Context, spec *job.OCR2OracleSpec, rc
...
+	onchainSigningStrategy := OCR2OnchainSigningStrategy{}
+	err = json.Unmarshal(spec.OnchainSigningStrategy.Bytes(), &onchainSigningStrategy)
+	if err != nil {
+		return err
+	}
+	pk, err := onchainSigningStrategy.PublicKey()
+	if err != nil {
+		return err
+	}
+	fmt.Println("public key", pk)
```
Comment (cedric-cordenier) [inline core/services/ocr2/validate/validate.go:236]: Same here

**Review by cedric-cordenier — APPROVED** [review]
Looks good!

**Review by cedric-cordenier — APPROVED** [review]

**Review by cedric-cordenier — DISMISSED** [review]

**Comment by vyzaldysanchez** [comment]
@cedric-cordenier I forgot to add the changes that would store the values into the DB, they are now added.

**Review by cedric-cordenier — APPROVED** [review]

**Review by george-dorin — APPROVED** [review]

**Pull request merged**